packages/tests/unit-tests/test-cases/expected/react/component-input-defaults.tsx

```diff
@@ -143,52 +142,30 @@ export default function Widget(props: typeof WidgetProps & RestProps) {
     [props]
   );
   const __nestedChildren = useCallback(
-    function __nestedChildren<T>(): T[] {
+    function __nestedChildren(): Record<string, any> {
       const { children } = props;
       return __collectChildren(children);
     },
     [props.children]
   );
   const __getNestedTexts2 = useCallback(
     function __getNestedTexts2(): typeof TextsProps {
-      const nested = __nestedChildren<typeof TextsProps & { __name: string }>()
-        .filter((child) => child.__name === "texts2")
-        .map((n) => {
-          if (
-            !Object.keys(n).some(
-              (k) => k !== "__name" && k !== "__defaultNestedValues"
-            )
-          ) {
-            return (n as any)?.__defaultNestedValues || n;
-          }
-          return n;
-        });
+      const nested = __nestedChildren();
       return props.texts2
         ? props.texts2
-        : nested.length
-        ? nested?.[0]
+        : nested.texts2
+        ? nested.texts2?.[0]
```

Comment:
Why ?.[0] is needed here?

Reply:
Because is object (@Nested() texts2?: TextsProps) but nested.texts2 is array